torch/distributed/distributed_c10d.py

```diff
@@ -298,6 +298,8 @@ def __getattribute__(self, key):
 # For a pg, it is a map from ProcessGroup to BackendConfig
 _pg_backend_config: Dict[ProcessGroup, str] = {}
 _group_count = 0
+_tags_to_pg: Dict[str, List[ProcessGroup]] = {}
```

Comment:
just to clarify: we can specify the same tag for multiple process gps?
nit: I would rename to `_tag_to_pgs` to be more clear about the mapping.

Reply:
You can have the same tag for PGs with different set of ranks.
But if you try to create a PG with the same tag and set of ranks, you'll get the original one instead.